docs(fields): add missing validation methods in jsdoc and improve fields documentation (#958)
Refinitiv/refinitiv-ui: documents/src/pages/elements/email-field.md

`ef-email-field` has validation logic similar to a native input. When a user types the invalid value into the control, error style will be shown to notify the user. However, if the control is being initialised with an invalid value, `reportValidity()` must be called to ensure the error style is applied.

::
```javascript
::email-field::
const emailField = document.querySelector('ef-email-field');
const button = document.querySelector("ef-button");

button.addEventListener('tap', () => {
  emailField.reportValidity();
})
```
```html
<ef-email-field value="1"></ef-email-field>
<ef-button>Submit</ef-button>
```
::

### Displaying error messages

> Whenever input is invalid, the error attribute will be added to the element. You can use the `error` property to check if input is currently in the error state.

Reviewer: ```suggestion
Whenever input is invalid, the `error` attribute will be added to the element. You can use the `error` property to check if input is currently in the error state. Note that, if input is initialised with invalid value, `reportValidity()` must be called first as described in [Input Validation](/elements/email-field#input-validation).
```

See the [Input Length](/elements/email-field#input-length) example below for more detail.

## Input length
The `maxlength` attribute limits the number of characters that users can type into the input and the `minlength` attribute is used to set the minimum of characters required. `ef-email-field` will show error styles if the condition is not met.

> @> Constraint validation is only applied when the value is changed by the user. [See input email](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/email#maxlength).

Reviewer: Let's clarify that only these 2 validations require user-triggered changes.
```suggestion
@> `maxlength` and `minlength` constraint validations are only applied when the value is changed by the user. [See input email](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/email#maxlength).
```

::
```javascript
::email-field::
const emailField = document.querySelector("ef-email-field");
const errorText = document.getElementById('error-text');

emailField.addEventListener("blur", (event) => {
  if (emailField.error) {
    errorText.innerHTML = "Must be in standard email format with between 8-14 characters.";
  }
  else {
    errorText.innerHTML = "";

>   }

Reviewer: ```suggestion
  errorText.innerHTML =  emailField.error
    ? "Must be in standard email format with between 8-14 characters."
    :  "";
```

```javascript
const emailField = document.querySelector("ef-email-field");
const errorText = document.getElementById('error-text');

emailField.addEventListener("blur", (event) => {
  if (emailField.error) {
    errorText.innerHTML = "Must be in standard email format with between 8-14 characters.";
  }
  else {
    errorText.innerHTML = "";

>   }

Reviewer: let's simplify with this.  Not sure why suggestion doesn't work here.
```  
  errorText.innerHTML =  emailField.error
    ? "Must be in standard email format with between 8-14 characters."
    :  "";
```

});

emailField.addEventListener("input", () => {
  if (!emailField.error) {
    errorText.innerHTML = "";
  }
});
```

```typescript
import { EmailField } from '@refinitiv-ui/elements/email-field';

> const emailField = document.querySelector("ef-email-field") as EmailField;

Reviewer: ```suggestion
const emailField = document.querySelector<EmailField>("ef-email-field");
```

const errorText = document.getElementById('error-text');

emailField?.addEventListener("blur", () => {
  if (!errorText) {
    return;
  }
  if (emailField?.error) {
    errorText.innerHTML = "Must be in standard email format with between 8-14 characters.";
  }
  else {
    errorText.innerHTML = "";

>   }

Reviewer: let's simplify with this.  Not sure why suggestion doesn't work here.
```  
  errorText.innerHTML =  emailField.error
    ? "Must be in standard email format with between 8-14 characters."
    :  "";
```

```javascript
::email-field::
const emailField = document.querySelector("ef-email-field");
const errorText = document.getElementById("error-text");

emailField.addEventListener("blur", () => {
  if (emailField.error) {
    errorText.innerHTML = "Email must end with '@mail.com'.";
  }
  else {
    errorText.innerHTML = "";

>   }

Reviewer: let's simplify with this.  Not sure why suggestion doesn't work here.
```  
  errorText.innerHTML =  emailField.error ? "Email must end with '@mail.com'." :  "";
```

```javascript
const emailField = document.querySelector("ef-email-field");
const errorText = document.getElementById("error-text");

emailField.addEventListener("blur", () => {
  if (emailField.error) {
    errorText.innerHTML = "Email must end with '@mail.com'.";
  }
  else {
    errorText.innerHTML = "";

>   }

Reviewer: let's simplify with this.  Not sure why suggestion doesn't work here.
```  
  errorText.innerHTML =  emailField.error ? "Email must end with '@mail.com'." :  "";
```

});

emailField.addEventListener("input", () => {
  if (!emailField.error) {
    errorText.innerHTML = "";
  }
});
```

```typescript
import { EmailField } from "@refinitiv-ui/elements/email-field";

> const emailField = document.querySelector("ef-email-field") as EmailField;

Reviewer: ```suggestion
const emailField = document.querySelector<EmailField>("ef-email-field");
```

const errorText = document.getElementById("error-text");

emailField?.addEventListener("blur", () => {
  if (!errorText) {
    return;
  }
  if (emailField?.error) {
    errorText.innerHTML = "Email must end with '@mail.com'.";
  }
  else {
    errorText.innerHTML = "";

>   }

Reviewer: ```suggestion
  errorText.innerHTML =  emailField?.error ? "Email must end with '@mail.com'." :  "";
```

::email-field::
const emailField = document.querySelector('ef-email-field');
const emailList = document.getElementById('email-added');
const errorText = document.getElementById("error-text");

emailField.addEventListener('icon-click', (e) => {
  if (!emailField.error && emailField.value.length > 0) {
    emailList.innerHTML = emailField.value + " is added.";
  }
});
emailField.addEventListener("blur", () => {
  if (emailField.error) {
    errorText.innerHTML = "Invalid email format.";
    emailList.innerHTML = "";
  }
  else {
    errorText.innerHTML = "";

>   }

Reviewer: let's simplify with this:
```
errorText.innerHTML = emailField.error ? "Invalid email format." : "";
```
